src/parser/mod.rs

```diff
@@ -49,6 +49,7 @@ pub trait Parser: ParserConfig {
 
     /// Parses a macro name, returning the name and the extracted variables
     fn parse_name<'a>(&self, mut input: &'a str) -> Result<(String, Vec<&'a str>), ParseError> {
+        let original = input.to_string();
```

Comment:
I think it would be better to defer to heap allocation of `.to_string()` by just `Copy`ing the string slice, i.e. `let original = input;` here and `return Err(ParseError::UnclosedVariableError(original.to_owned()));` below, to avoid the overhead in the common case. (Note that shared references `&T` are always `Copy`.) (Also note that `.to_string()` is used to get a stringly-typed representation of some type `T: Display` whereas you already have something stringly-typed and just want an owned version of it, hence `.to_owned()` better expresses the intent here and `.to_string()` doing effectively the same is pretty much coincidence.)

Reply:
@adamreichold Thanks, changed accordingly.